src/builtins/Transformers.jl

```diff
@@ -516,58 +518,50 @@ MLJBase.output_is_multivariate(::Type{<:OneHotEncoder}) = true
 
 ## Imputer
 """
-    fillImputer -   imputes by applying function to non missing values
+    FillImputer -   imputes by applying function to non missing values
                     default for continuous(median), CategoricalArray(most common),
                         Count (rounded median)
 
 
 """
-
-function commonCategoryFiller(vec::CategoricalArray)
-    ftab= freqtable(vec)
-    return names(ftab)[1][findmax(ftab)[2]]
-end
-
-function basetype(A::B) where {T,N,B<:AbstractArray{T,N}}
-    T
-end
-
-
-
-
-mutable struct fillImputer <: Unsupervised
+mutable struct FillImputer <: Unsupervised
     features::Vector{Symbol}
     continuous_fill::Union{Function}
     count_fill::Union{Function}
     categorical_fill::Union{Function}
     allowed_scitypes::Array{Any}
 
 end
-function common_cat_filler(vec::CategoricalArray)
+
+FillImputer(;features=Symbol[], continuous_fill=median,
+    count_fill=basic_count_fill ,categorical_fill=x->common_cat_filler(x),
+    allowed_scitypes=[Union{Continuous,Missing},Union{Count,Missing},Union{Multiclass,Missing}])= FillImputer(features,
+        continuous_fill,count_fill,categorical_fill,allowed_scitypes)
+
+
+function commonCategoryFiller(vec::CategoricalArray)
     ftab= freqtable(vec)
     return names(ftab)[1][findmax(ftab)[2]]
 end
 
 
-function basetype(A::B) where {T,N,B<:AbstractArray{T,N}}
-    T
+function common_cat_filler(vec::CategoricalArray)
+    return StatsBase.mode(vec)
 end
 
+
+
 function basic_count_fill(v)
-    round(basetype(v),median(v))
+    round(eltype(v),median(v))
 end
 
-fillImputer(;features=Symbol[], continuous_fill=median,
-    count_fill=basic_count_fill ,categorical_fill=x->common_cat_filler(x),
-    allowed_scitypes=[Union{Continuous,Missing},Union{Count,Missing},Union{Multiclass,Missing}])= fillImputer(features,
-        continuous_fill,count_fill,categorical_fill,allowed_scitypes)
 
 
-mutable struct fillImputerResult <: Unsupervised
+mutable struct FillImputerResult <: Unsupervised
     features::Vector{Symbol}
 end
 
```

Comment:
Do you really need a struct for this? You're just wrapping one field. In any case, drop `<: Unsupervised` as this is not a model, right?